src/__tests__/type-modifiers.js

```diff
@@ -0,0 +1,321 @@
+import React from 'react'
+import {render} from '@testing-library/react'
+import userEvent from '../../src'
+
+// Note, use the setup function at the bottom of the file...
+// but don't hurt yourself trying to read it 😅
+
+// keep in mind that we do not handle modifier interactions. This is primarily
+// because modifiers behave differently on different operating systems.
+// For example: {alt}{backspace}{/alt} will remove everything from the current
+// cursor position to the beginning of the word on Mac, but you need to use
+// {ctrl}{backspace}{/ctrl} to do that on Windows. And that doesn't appear to
+// be consistent within an OS either 🙃
+// So we're not going to even try.
+
+// This also means that '{shift}a' will fire an input event with the shiftKey,
+// but will not capitalize "a".
+
+test('{esc} triggers typing the escape character', async () => {
+  const {element: input, getEventCalls} = setup('input')
+
+  await userEvent.type(input, '{esc}')
+
+  expect(getEventCalls()).toMatchInlineSnapshot(`
```

Comment:
This is such a slick method of testing. I ran into this same problem where tests became super hairy and difficult to reason about, but this is awesome :exploding_head: